coreos-installer

```diff
@@ -626,6 +630,65 @@ log() {
     echo "$1" >&2
 }
 
+#########################################################
+#Update bootloader if needed
+#########################################################
+write_bootloader() {
+    if [[ "$arch" == "s390x" ]]; then
+        log "Updating zipl"
+        mkdir -p /mnt/boot_partition
+        mount_boot_partition /mnt/boot_partition
+        trap 'umount /mnt/boot_partition; trap - RETURN' RETURN
+
+        blsfile=$(find /mnt/boot_partition/loader/entries/*.conf)
+        echo "$(grep options $blsfile | cut -d' ' -f2-) $(cat /tmp/s390x_opts)" > /tmp/zipl_prm
+        zipl --verbose -p /tmp/zipl_prm -i $(find /mnt/boot_partition/ostree/ -name "*vmlinuz*") -r $(find /mnt/boot_partition/ostree/ -name "*initramfs*") --target /mnt/boot_partition
+        if [[ $? -ne 0 ]]; then
+            log "failed updating bootloder"
+            exit 1
+        fi
+        # this allows a reboot into new installed disk
+        chreipl "${DEST_DEV}"
+    fi
+}
+#########################################################
+#Handle s390x zFCP SCSI devices
+#########################################################
+write_image_to_zfcp_disk() {
+    log "Extracting disk image"
+    # extract and remove downloaded image
+    $DECOMPRESSION_TOOL /mnt/dl/imagefile.compressed > /mnt/dl/imagefile
+
+    (while IFS='' read -r line || [ -n "$line" ]
+    do
+        if [[ $line =~ "start=" ]]
+        then
+            # we save the start= and size= here, then dd the partitions later
+            printf "%s:%s\n" \
+                "$(echo $line | cut -d: -f 2 | cut -d, -f 1 | tr -d ' ')" \
+                "$(echo $line | cut -d: -f 2 | cut -d, -f 2 | tr -d ' ')" \
+                >> /tmp/sfdisk_info
+        elif ! [[ $line =~ "label:" || $line =~ "label-id:" || $line =~ "unit:" ]]
+        then
+            continue
+        fi
+        echo $line
+    done <<< $(sfdisk -d /mnt/dl/imagefile)
+    ) | sfdisk "${DEST_DEV}"
+
+    while IFS='' read -r line || [ -n "$line" ]
+    do
+        eval ${line%%:*} ${line##*:}
+        # NOTE: we hardcode block size to 512 for current zFCP SCSI devices
+        dd if=/mnt/dl/imagefile bs=512 iflag=fullblock \
```

Comment:
Is it because there is no `oflag=direct` or a `$ sync` call so that lsblk fails to list UUID ?